src/k8sconfiguration/azext_k8sconfiguration/custom.py

```diff
@@ -229,7 +250,10 @@ def __validate_url_with_params(repository_url, ssh_private_key_set, known_hosts_
 
 
 def __validate_known_hosts(knownhost_data):
-    knownhost_str = __from_base64(knownhost_data).decode('utf-8')
+    try:
+        knownhost_str = __from_base64(knownhost_data).decode('utf-8')
+    except Exception as ex:
+        raise CLIError('Error! ssh known_hosts is not a valid utf-8 base64 encoded string') from ex
```

Comment:
Should CLIError be deprecated? @yungezz

Reply:
Is there a more up to date error to throw?

Reply:
Please check the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md)

Reply:
Updated